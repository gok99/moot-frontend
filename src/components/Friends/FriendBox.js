import React, { Component } from 'react';
import { Row, Col } from 'react-bootstrap';

import '../Styles/styles.css';
import logo_temp from '../../assets/logo_temp.png';

class FriendBox extends Component {
  render() {
    return (
      <div className="contentbox spacedbox">
        <Row>
          <Col xs={2} className="d-flex justify-content-center">
            <img className="previewpic" src={logo_temp} alt="Profile" />
          </Col>
          <Col xs={6} className="d-flex justify-content-center">
            <p className="friendboxtext">INSERT HYPOTHETICAL FRIEND NAME HERE</p>
          </Col>
          <Col xs={4} className="d-flex justify-content-center">
            <a href="https://t.me/moot_chat1_bot" target="_blank" rel="noopener noreferrer" className="mootbutton smallmedbutton">
              Chat -&#62;
            </a>
          </Col>
        </Row>
      </div>
    )
  }
}

export default FriendBox;