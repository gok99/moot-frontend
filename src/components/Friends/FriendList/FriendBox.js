import React from 'react';
import { Row, Col } from 'react-bootstrap';

import '../../Styles/styles.css';
import '../friends.css';

const FriendBox = (props) => {
  const fTeleUser = props.fTeleUser;
  const fUsername = props.fUsername;
  return (
    <div className="b-friendbox">
      <Row>
        <Col xs={6} className="mt-1">
          <Row>
            <p className="text-friendbox heavy">{fUsername}</p><br /><br />
          </Row>
          <Row>
            <p className="text-friendbox light">@{fTeleUser}</p>
          </Row>
        </Col>
        <Col xs={4} className="d-flex justify-content-center">
          <a href={"https://t.me/"+fTeleUser} target="_blank" rel="noopener noreferrer">
            Chat &#62;&#62;&#62;
          </a>
        </Col>
      </Row>
    </div>
  );
};

export default FriendBox;