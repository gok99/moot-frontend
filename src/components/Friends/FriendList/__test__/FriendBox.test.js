import React from 'react';
import ReactDOM from 'react-dom';
import FriendBox from '../FriendBox';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<FriendBox fUid={0} fUsername={""} fTeleUser={""} onViewPost={(fUid, fUsername) => (event) => { event.preventDefault(); }} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<FriendBox fUid={0} fUsername={""} fTeleUser={""} onViewPost={(fUid, fUsername) => (event) => { event.preventDefault(); }} />).toJSON();
  expect(tree).toMatchSnapshot();
});