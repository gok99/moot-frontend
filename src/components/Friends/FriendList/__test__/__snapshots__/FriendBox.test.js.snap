// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="b-friendbox mb-2 row"
>
  <div
    className="b-text-friendbox col"
  >
    <div
      className="row"
    >
      <p
        className="text-friendbox heavy"
      >
        
      </p>
    </div>
    <div
      className="row"
    >
      <p
        className="text-friendbox light"
      >
        @
        
      </p>
    </div>
  </div>
  <div
    className="d-flex justify-content-end col-md-auto"
  >
    <button
      className="btn-friendchat btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      View 
      
      's Posts
    </button>
  </div>
  <div
    className="d-flex justify-content-end col-md-auto"
  >
    <a
      className="btn-friendchat btn btn-primary"
      href="https://t.me/"
      onClick={[Function]}
      onKeyDown={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Chat with 
      
    </a>
  </div>
</div>
`;
